### PR TITLE
[GEARPUMP-197] fix busy loop in FetchThread

### PR DESCRIPTION
**Pull request by manuzhang**

### CODE REVIEW

**Comment by huafengw** [comment]
+1

File: external/kafka/src/main/scala/org/apache/gearpump/streaming/kafka/lib/source/consumer/FetchThread.scala [hunk 1]
```diff
@@ -133,19 +133,21 @@ private[kafka] class FetchThread(
 
   /**
    * fetch message from each TopicAndPartition in a round-robin way
+   *
+   * @return whether to fetch more messages
```
Comment (kkasravi) [inline external/kafka/src/main/scala/org/apache/gearpump/streaming/kafka/lib/source/consumer/FetchThread.scala]: akka-streams has this GraphStage called InterLeave where upstreams are cycled and N messages are pulled from each one. I'm wondering if we should expose a policy related to fetch messages which is now round-robin.

**Comment by kkasravi** [comment]
+1 though see my comment on round-robin fetching of messages

File: external/kafka/src/main/scala/org/apache/gearpump/streaming/kafka/lib/source/consumer/FetchThread.scala (hunk 1 above)
Comment (manuzhang) [inline external/kafka/src/main/scala/org/apache/gearpump/streaming/kafka/lib/source/consumer/FetchThread.scala]: Could you create a jira ?

**Pull request closed without merging**